### DISCUSSION TITLE
HAN Convolution usage without meta paths

### DISCUSSION THREAD

**Discussion by mfaizan-ai** [pyg-team/pytorch_geometric#6552] [General]
Dear Author, 
I am trying to implement [HANConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.HANConv) with the meta paths as mentioned in its paper. We are using k hops metrics instead. I want to know how to use it with K metrics. Like we generated K metrics based on the neighborhood of node by calculating length of the node with its neighbors; then, if the length is less than certain threshold we kept the node as neighbor otherwise it is not. There is no option in the HANConv args to accept something like that, it just accept meta paths i guess. I am trying to use it with IMDB dataset.

**Comment by rusty1s** [comment]
Can you clarify what you mean by "K metrics"?

**Reply by malikhizar** [reply]
@rusty1s **`K`** matrix is a matrix of size `|N| x |N|` where `N` is number of nodes. **`K`** has entries `1` and `0` only. If the distance between any two nodes for instance, node `a ∈ N`, and node `b ∈ N` is equal to or greater than a certain threshold, we consider `b` as a neighbor of node `a` and put the entry `1` in the matrix for `a` and `b` intersection box, and `0` otherwise. Now we want to use the nodes' neighborhood matrix **`K`** instead of meta-paths in [HANConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.HANConv) method.

**Reply by rusty1s** [reply]
Any reason you want to use `HANConv` in particular (since it is designed for heterogeneous graphs)? IMO, you can just construct `edge_index` from `K` and input that into any GNN layer:
```python
edge_index = K.nonzero().t()
```

**Reply by mfaizan-ai** [reply]
@rusty1s @khizer-hayat could we calculate shortest distance between nodes in a heterogeneous graph like IMDB?
